src/libsyncengine/syncpal/tmpblacklistmanager.cpp

```diff
@@ -89,6 +89,17 @@ void TmpBlacklistManager::blacklistItem(const NodeId &nodeId, const SyncPath &re
     }
 
     insertInBlacklist(nodeId, side);
+
+    std::list<NodeId> toBeRemoved;
+    for (const auto &errorInfo: errors) {
+        if (Utility::isDescendantOrEqual(errorInfo.second.path, relativePath) && errorInfo.first != nodeId) {
+            toBeRemoved.push_back(errorInfo.first);
+        }
+    }
+
+    for (const auto &id: toBeRemoved) {
+        removeItemFromTmpBlacklist(id, side);
+    }
```

Comment:
Is it really necessary to clean up the tmp blacklist?
Isn't it more efficient to add everything in it and just check when needed if an item or any ancestor appears in the blacklist?

Reply:
Moreover, if we have A/B and we blacklist B then A, then we unblacklist A, B will no longer be blacklisted.

Reply:
@ClementKunz Since the blacklist is stored in the database, I wanted to keep it as short as possible. For example, if a user encounters an access denial on a folder containing 1,000 files, this would create 1,000 temporarily blacklisted items.
Each update to the temporary blacklist will clear the database and reinsert all elements, plus any new additions or minus any removals. This means approximately 1,000² insertions and deletions in the database just for a lost of right a big folder.
Similarly, searching will be faster if we only keep the parent folders rather than every individual file.

@ChristopheLarchier
I assumed that since the temporary blacklist is for temporary errors and would automatically unblacklist items after one hour, it wouldn’t be a major issue to unblacklist children if the parent is unblacklisted. If a child was blacklisted for another reason, it would be re-blacklisted in the next sync.

It may be a bit optimistic on my part. Please let me know what you think.

Reply:
Ok for me 👍